### PR TITLE
Add Functionality for Closing Stream (w/o Cancel) Allow Transcription

### PR DESCRIPTION
**Pull request by dvonthenen**
Completion

## Proposed changes

Address: https://github.com/deepgram/deepgram-dotnet-sdk/issues/341

The `Stop()~ function has been changed to no longer use cancel to trigger the stop, but to fire the `Close` message with a small delay, does a "best effort" to allow any final transcription to get passed back to the client.

This also changes the TTS WS example to use linear16 which is more inline with examples in other SDKs.

## Types of changes

What types of changes does your code introduce to the community .NET SDK?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update or tests (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [x] I have lint'ed all of my code using repo standards
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

## Further comments

NA

## Summary by CodeRabbit

- **New Features**
	- Enhanced WebSocket client interfaces with new optional parameters for improved control over connection closure.
	- Updated audio output handling to support WAV format and include a WAV header.

- **Bug Fixes**
	- Improved error handling and logging during WebSocket connection and disconnection processes.

- **Documentation**
	- Added comments to clarify the functionality of new parameters and methods.

### CODE REVIEW

File: examples/text-to-speech/websocket/simple/Program.cs [hunk 1]
```diff
@@ -41,9 +44,72 @@ static async Task Main(string[] args)
...
+                        wavHeader[39] = 0x61; // a
+                        wavHeader[40] = 0x00; // Placeholder for data chunk size (will be updated later)
+                        wavHeader[41] = 0x00; // Placeholder for data chunk size (will be updated later)
+                        wavHeader[42] = 0x00; // Placeholder for data chunk size (will be updated later)
+                        wavHeader[43] = 0x00; // Placeholder for data chunk size (will be updated later)
+
+                        writer.Write(wavHeader);
+                        appendWavHeader = false;
+                    }
+                }
```
[the remark this answers is not in the document]
Comment (dvonthenen) [reply examples/text-to-speech/websocket/simple/Program.cs:108]: This is just an example, doesnt need to be perfect.

File: Deepgram/Clients/Listen/v1/WebSocket/Client.cs [hunk 2]
```diff
@@ -842,25 +842,16 @@
...
-                {
-                    _clientWebSocket.SendAsync(new ArraySegment<byte>([0]), WebSocketMessageType.Binary, true, cancelToken.Token)
-                        .ConfigureAwait(false);
-                }
+                SendClose(nullByte);
             }
 
+            // small delay to wait for any final transcription
+            await Task.Delay(100, cancelToken.Token).ConfigureAwait(false);
+
```
[the remark this answers is not in the document]
Comment (dvonthenen) [reply Deepgram/Clients/Listen/v1/WebSocket/Client.cs:854]: Noted.

File: examples/text-to-speech/websocket/simple/Program.cs [hunk 3]
```diff
@@ -41,9 +44,72 @@
...
+
                 if (e.Stream != null)
                 {
-                    using (BinaryWriter writer = new BinaryWriter(File.Open("output.mp3", FileMode.Append)))
+                    using (BinaryWriter writer = new BinaryWriter(File.Open("output.wav", FileMode.Append)))
                     {
                         writer.Write(e.Stream.ToArray());
```
[the remark this answers is not in the document]
Comment (dvonthenen) [reply examples/text-to-speech/websocket/simple/Program.cs:114]: This is just an example, doesnt need to be perfect.

File: examples/text-to-speech/websocket/simple/Program.cs [hunk 4]
```diff
@@ -75,7 +141,11 @@
             }));
 
             // Start the connection
-            var speakSchema = new SpeakSchema();
+            var speakSchema = new SpeakSchema()
+            {
+                Encoding = "linear16",
+                SampleRate = 48000
+            };
```
[the remark this answers is not in the document]
Comment (dvonthenen) [reply examples/text-to-speech/websocket/simple/Program.cs:148]: This is just an example, doesnt need to be perfect.

**Review by jpvajda — APPROVED** [review]

**Pull request merged**